code/modules/modular_computers/networking/machinery/_network_machine.dm

```diff
@@ -5,17 +5,45 @@
 	var/main_template = "network_mainframe.tmpl"
 	var/network_device_type =  /datum/extension/network_device
 	var/error
-	
+
 	var/initial_network_id
 	var/initial_network_key
 	var/lateload
 
+	var/produces_heat = TRUE	// If true, produces and is affected by heat.
+	var/inefficiency = 0.12		// How much power is waste heat.
+	var/heat_threshold = 358.15	// At what temperature the machine will lock up.
```

Comment:
Please use temperatures of the form `T0C + X` or `X CELSIUS` to improve readability.